### ISSUE TITLE
New state ignored when same reference returned from reducer

### ISSUE THREAD

**Issue by mightyiam** [cyclejs/cyclejs#889]
It seems that when a reducer returns *not* a new object, but a reference to the same object it received, that reduction is ignored and `sources.state.stream` does not emit. Am I right about this?

I'm OK with this behavior, as long as it is documented. And perhaps a good exception would help there, as well.
"The following reducer returned a reference to the same state object that it had received. That is not supported. Please modify the reducer to return a new state object."

Pull request welcome?

**Versions of packages used:**

```
├─┬ @cycle/dom@22.3.0
│ ├── @cycle/run@5.2.0 deduped
├─┬ @cycle/run@5.2.0
├─┬ @cycle/state@1.2.0
│ ├─┬ @cycle/isolate@5.0.0
│ │ ├── @cycle/run@5.2.0 deduped
│ ├── @cycle/run@5.2.0 deduped
```

**Comment by sarimarton** [comment]
I disagree with the need for having an exception for this, but a more detailed documentation is of course can't do any harm.

I guess this dropRepeats() is responsible for the behavior:

https://github.com/cyclejs/cyclejs/blob/243cd2534ff2728ad8c84b1ce9d85fe13e61fd8b/state/src/StateSource.ts#L94

**Comment by jvanbruegge** [comment]
Yeah, this is intentional and I also don't think adding a warning is necessary, but it would be nice to document this more explicitly

**Comment by mightyiam** [comment]
Could you please explain the intention? I can't think of one.
